AirportTicketBookingExercise/Logic/Service/BookingService.cs

```diff
@@ -0,0 +1,174 @@
+        {
+            return _bookingRepository.DeleteBooking(bookingId);
+        }
+
+        public bool UpdateBookingClass(int bookingId, BookingClass newClass)
+        {
+            return _bookingRepository.UpdateBookingClass(bookingId, newClass);
+        }
+
+        public bool Cancel(int bookingId, User loggedInUser)
+        {
+            try
+            {
+                if (!_bookingRepository.IsBookingValidById(bookingId, loggedInUser.UserId))
+                    return false;
+                bool isSuccess = _bookingRepository.DeleteBooking(bookingId);
+                if (!isSuccess)
+                    return false;
+                else return true;
+            }
+            catch
+            {
+                return false;
+            }
+        }
+
+        public bool Modify(int bookingId, BookingClass bookingClass, User loggedInUser)
+        {
+            if (!_bookingRepository.IsBookingValidById(bookingId, loggedInUser.UserId))
+                return false;
+            _bookingRepository.UpdateBookingClass(bookingId, bookingClass);
+            return true;
+        }
+
+        public bool PassengerBookFlight(int flightId, BookingClass bookingClass, User loggedInUser)
+        {
+            try
+            {
+                Flight? flight = _flightRepository.GetFlight(flightId);
+                if (flight == null || flight.SeatsAvailable >= flight.SeatCapacity)
+                    return false;
+                decimal price = bookingClass switch
+                {
+                    BookingClass.First => flight.FirstClassPrice,
+                    BookingClass.Business => flight.BuisnessPrice,
+                    BookingClass.Economy => flight.EconomyPrice,
+                    _ => 0
+                };
+                if (price == 0)
+                    return false;
+                var Booking = new Booking
+                {
+                    FlightId = flightId,
+                    PassengerId = loggedInUser.UserId,
+                    BookingClass = bookingClass
+                };
+                _bookingRepository.CreateBooking(Booking);
+                _flightRepository.AddPassengerToSeat(flight);
+                return true;
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Error while booking: {ex.ToString()}");
+                return false;
+            }
+        }
+
+        public List<Booking> FilterBookings(string[] filterInput)
+        {
+            BookingFilter query = BookingFilters.Parse(filterInput.Skip(1).ToArray());
+            List<Booking> bookingResults = new List<Booking>();
+
+            var bookings = _bookingRepository.GetAllBookings();
+            var flights = _flightRepository.GetFlights().ToDictionary(f => f.FlightId);
+            var passengers = _userRepository.GetUsersByType(UserType.Passenger).ToDictionary(u => u.UserId);
+
+            bookingResults = bookings.Where(b =>
```

Comment:
think about rewriting this using joins and just one straightforward query, is a bit odd to in the where statement all this logic, can be done with multiple where statements